website/docs/contributing/ADRs/front-end/sdk-generator.md

```diff
@@ -0,0 +1,33 @@
+---
+title: "ADR: OpenAPI SDK Generator"
+---
+
+## Background
+
+Our current frontend setup we have a lot of code that can be generated out of OpenAPI schema. Types have not been updated in a while, and even some new features rely on hard-coded types instead of auto-generated files in `src/openapi`. Fetchers and actions in the frontend involves custom-built code that is grouped in the `src/hooks/api` folder. There is a separation between getters and actions. Getters use the SWR library. API actions (POST/PUT/DELETE) are grouped by feature and are exposed to components reliant on `useAPI` hook.
+
+SDK generated out of it will be better then what we have right now. It will help to reduce the risk of duplication and inconsistencies in our SDK, as it will be generated from a single source of truth.
```

Comment:
```suggestion
An SDK generated out of it will be better than what we have right now. It will help reduce the risk of duplication and inconsistencies in our SDK, as it will be generated from a single source of truth.
```